### ISSUE TITLE
Messages arrive out of order after switching from Desktop app to Android app

### ISSUE THREAD

**Issue by goaaats** [signalapp/Signal-Android#10700]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
When switching from a conversation on the Desktop app(Windows 10, v1.39.5) to Android, messages will slowly catch up, but end up displaying out of order.

The correct order is not restored after restarting the app.

### Steps to reproduce
- Start a lengthier conversation with a contact on Desktop
- Wait a few minutes, keep the Android phone locked
- Tap the notification on the Android app immediately(this was important for reproduction in my experience) after unlocking and watch messages replay out of order

**Actual result:** Messages do not arrive in the correct order, nearby messages get shuffled and the conversation is hard to follow
**Expected result:** Messages should display consistently in the same order on Android as they initially arrived in on the Desktop app

### Screenshots
Android:
![Screenshot_20210117-192223](https://user-images.githubusercontent.com/16760685/104852260-38074180-58fa-11eb-97a8-e973be1d05be.jpg)

Desktop
![DesktopScreenshot](https://user-images.githubusercontent.com/16760685/104852273-4ce3d500-58fa-11eb-8b18-a5fdd9967e9b.png)

You can see the difference by observing the cat sticker. This is just an example, the end of each "replay" will be jumbled up in various places if the bug was reproduced. I was able to get this to happen quite consistently.

### Device info

**Device:** OnePlus 7 Pro(GM1913)
**Android version:** 10 (OxygenOS 10.0.10.GM21BA)
**Signal version:** 5.2.1

### Link to debug log
Android app: https://debuglogs.org/020edaf31cff48e227f72f6827438d70808d527dbffaa52d5dc8e4dbb036083d
Desktop app: https://debuglogs.org/ac4dddce471e679b31e081a1c0a4b8116ac9b9db5977a9d4ff5b4737326aa175

**Comment by simonmueck** [comment]
Got the same issue (iPhone 6 & Windows 10)

**Comment by Leptopoda** [comment]
Hi there this is very much known and sort of accepted behaviour. The problem is that messages get displayed in order they are received by the other device rather than in order they where originally sent.
There are open feature requests about this in the [community forum](https://community.signalusers.com)
I suggest joining a discussion there as this is not a bug rather than a 'bad design decision'

I therefore suggest closing this issue
